examples/foundational/21-tavus-layer.py

```diff
@@ -0,0 +1,140 @@
+        )
+
+        room_url, conversation_id = TavusVideoService._initiate_conversation(
+            api_key=os.getenv("TAVUS_API_KEY"),
+            replica_id=os.getenv("TAVUS_REPLICA_ID"),
+            persona_id=os.getenv("TAVUS_PERSONA_ID", "pipecat0"),
+            custom_greeting='Hello, I am pipecat',
+
+        )
+
+        transport = DailyTransport(
+            room_url=room_url,
+            token=None,
+            bot_name="Pipecat bot",
+            params=DailyParams(
+                transcription_enabled=True,
+                vad_enabled=True,
+                vad_analyzer=SileroVADAnalyzer(),
+                vad_audio_passthrough=True,
+            ),
+        )
+
+        stt = DeepgramSTTService(api_key=os.getenv("DEEPGRAM_API_KEY"))
+
+        tts = CartesiaTTSService(
+            api_key=os.getenv("CARTESIA_API_KEY"),
+            voice_id="79a125e8-cd45-4c13-8a67-188112f4dd22",  # British Lady
+        )
+
+        llm = OpenAILLMService(model="gpt-4o-mini")
+
+        messages = [
+            {
+                "role": "system",
+                "content": "You are a helpful LLM in a WebRTC call. Your goal is to demonstrate your capabilities in a succinct way. Your output will be converted to audio so don't include special characters in your answers. Respond to what the user said in a creative and helpful way.",
+            },
+        ]
+
+        tma_in = LLMUserResponseAggregator(messages)
+        tma_out = LLMAssistantResponseAggregator(messages)
+
+        tavus = TavusVideoService(
+            conversation_id=conversation_id,
+            client=transport._client,
+            params=transport._params
+        )
+
+        pipeline = Pipeline(
+            [
+                transport.input(),  # Transport user input
+                stt,  # STT
+                tma_in,  # User responses
+                llm,  # LLM
+                tts,  # TTS
+                tavus, # Tavus output layer
+                # transport.output(),  # Transport bot output
+                tma_out,  # Assistant spoken responses
+            ]
+        )
+
+        task = PipelineTask(
+            pipeline,
+            PipelineParams(
+                allow_interruptions=True,
+                enable_metrics=True,
+                enable_usage_metrics=True,
+                report_only_initial_ttfb=True,
+            ),
+        )
+
+        @transport.event_handler("on_participant_joined")
+        async def on_participant_joined(transport: DailyTransport, participant: dict[str, Any]) -> None:
+            if participant.get("info", {}).get("userName", "") == persona_name:
+                transport._client._client.update_subscriptions(
+                    participant_settings={
+                        participant["id"]: {
+                            "media": { "microphone": "unsubscribed" },
+                        }
+                    }
+                )
```

Comment:
Is this to avoid getting audio from the Tavus persona? Maybe we pass a `unsubcriptions` in `DailyParams` so you can do:

```
        transport = DailyTransport(
            room_url=room_url,
            token=None,
            bot_name="Pipecat bot",
            params=DailyParams(
                transcription_enabled=True,
                vad_enabled=True,
                vad_analyzer=SileroVADAnalyzer(),
                vad_audio_passthrough=True,
                unsubscriptons=[{persona_name: ("microphone")}]
            ),
        )
```

I'll see if I can add that.

Reply:
or actually better... just exposing `update_subscriptions`

Reply:
The only consideration there is that our participant may not be in the room when `transport` is being defined. Does a subscription persist for any new participants that join after the mic was unsubscribed to:

- Tavus replica hasn't joined yet
- Transport object initialized (we tell here to ignore any user with name 'Tavus')
- This should not cause a no-op but rather act as a blacklist for any user who may join with the name 'Tavus' in the future
- Tavus joins afterwards, Pipecat needs to automatically unsubscribe from Tavus' microphone

Reply:
I just added https://github.com/pipecat-ai/pipecat/pull/643. So you can call the public `transport.update_subscriptions()`.

Reply:
Done. I pulled your branch `aleix/daily-update-subscriptions` into mine and am directly calling `transport.update_subscriptions()` instead of `transport._client.client.update_subscriptions()`